pkg/config/authorization/opa.go

```diff
@@ -74,24 +76,45 @@ func (opa *OPA) Call(pipeline common.AuthPipeline, ctx context.Context) (bool, e
 		results, err := opa.policy.Eval(opa.opaContext, options)
 
 		if err != nil {
-			return false, err
+			return nil, err
 		} else if len(results) == 0 {
-			return false, fmt.Errorf(invalidOPAResponseErrorMsg)
-		} else if allowed := results[0].Bindings["allowed"].(bool); !allowed {
-			return false, fmt.Errorf(unauthorizedErrorMsg)
+			return nil, fmt.Errorf(invalidOPAResponseErrorMsg)
+		} else if allowed := results[0].Bindings["allow"].(bool); !allowed {
+			return nil, fmt.Errorf(unauthorizedErrorMsg)
 		} else {
-			return true, nil
+			return results[0].Bindings, nil
 		}
 	}
 }
 
 func (opa *OPA) precompilePolicy() error {
 	policyName := fmt.Sprintf(`authorino.authz["%s"]`, opa.policyUID)
 	policyContent := fmt.Sprintf(policyTemplate, policyName, opa.Rego)
-	regoQuery := rego.Query("allowed = data." + policyName + ".allow")
-	regoModule := rego.Module(opa.policyUID+".rego", policyContent)
+	policyFileName := opa.policyUID + ".rego"
+
+	var module *opaParser.Module
+	rules := make(map[string]interface{})
```

Comment:
The "allow" query should always be injected or runtime errors may happen:

```
panic: interface conversion: interface {} is nil, not bool
```

Reply:
I guess this error would not happen due to https://github.com/Kuadrant/authorino/blob/b07bb9cc6f3d37e2fa9fd17000040ecee32ac650/pkg/config/authorization/opa.go#L21

On the other hand, if https://github.com/Kuadrant/authorino/pull/216#issuecomment-1047097541 is done, then yes. In case of looping through the list of rules to register all the queries is behind a conditional (i.e. only when `fuzzy: true`), then we'd need to ensure the "allow" query is always injected.

Reply:
Done.